gdal/apps/gdal_grid_lib.cpp

```diff
@@ -823,13 +832,27 @@ GDALDatasetH GDALGrid( const char *pszDest, GDALDatasetH hSrcDataset,
     if ( psOptions->pszSQL )
         nBands++;
 
-    // FIXME
-    int nXSize = psOptions->nXSize;
-    if ( nXSize == 0 )
-        nXSize = 256;
-    int nYSize = psOptions->nYSize;
-    if ( nYSize == 0 )
-        nYSize = 256;
+    int nXSize;
+    int nYSize;
+    if ( psOptions->dfXRes && psOptions->dfYRes )
+    {
+        nXSize = static_cast<int>(
```

Comment:
it would be more prudent to be able to catch wrong input values to store the result of the computation before the static_cast<int> in a double, dfXSize, and test that dfXSize >= 1 && dfXSize <= INT_MAX, before casting to nXSize. Same for y

Reply:
Added in 6641ab9